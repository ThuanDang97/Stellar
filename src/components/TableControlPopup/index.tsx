import React, { memo, useCallback } from 'react'

// Components
import { useTheme } from 'styled-components'
import { BoxShadowStyled, DividerStyled } from '@components/styled-components'
import Button from '@components/Button'

// Props type
import { TableControlPopupProps } from '@self-types/components/TableControlPopup.props'

// Utils
import { pxToRem } from '@utils/theme'

const TableControlPopup = React.forwardRef<
  HTMLInputElement,
  TableControlPopupProps
>(({ nameTable, handleEditTable, handleDeleteTable, idItem }, ref) => {
  const theme = useTheme()

  const onDeleteItem = useCallback(
    () => handleEditTable && handleEditTable(idItem),
    [handleEditTable, idItem],
  )

  return (
    <BoxShadowStyled
      display="flex"
      flexDirection="column"
      position="absolute"
      bgColor="white"
      width={110}
      shadowType="regular"
      border={`${pxToRem(theme.metrics.borderWidth.default)} solid ${
        theme.colors.silver
      }`}
      top={theme.metrics.dimensions.xs}
      nameTable={nameTable}
      mRight={135}
      ref={ref}
    >
      {nameTable === 'linkedBill' && (
        <>
          <Button
            variant="dark"
            width={theme.metrics.width.xs}
            height={theme.metrics.height.default}
            btnWidth={theme.metrics.width.full}
            cursor="pointer"
            onClick={onDeleteItem}
            data-testid="edit"
            title="Edit"
            textAlign="left"
            bgColorHover={theme.colors.ironLight}
          />
          <DividerStyled width={90} />
        </>
      )}
      <Button
        variant="dark"
        width={theme.metrics.width.xs}
        height={theme.metrics.height.default}
        btnWidth={theme.metrics.width.full}
        cursor="pointer"
        onClick={() => handleDeleteTable(idItem)}
        data-testid="edit"
        title="Delete"
        textAlign="left"
        bgColorHover={theme.colors.ironLight}
      />
    </BoxShadowStyled>
  )
})

export default memo(TableControlPopup)
