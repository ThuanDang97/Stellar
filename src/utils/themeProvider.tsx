import { ThemeProvider } from 'styled-components'
import { render } from '@testing-library/react'
import { theme } from '@themes/index'

export const renderWithTheme = (component: React.ReactNode) => {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>)
}
