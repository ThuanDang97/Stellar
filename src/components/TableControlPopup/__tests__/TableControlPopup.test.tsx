import { renderWithTheme } from '@utils/themeProvider'

// Components
import TableControlPopup from '@components/TableControlPopup'

describe('TableControlPopup Component', () => {
  it('renders table control popup ', () => {
    const { asFragment } = renderWithTheme(
      <TableControlPopup
        idItem=""
        handleEditTable={() => undefined}
        handleDeleteTable={() => undefined}
      />,
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
