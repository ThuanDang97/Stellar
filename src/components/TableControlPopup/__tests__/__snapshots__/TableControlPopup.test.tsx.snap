// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableControlPopup Component renders table control popup  1`] = `
<DocumentFragment>
  <div
    class="ViewStyled-sc-rqe5sa-0 ViewStyled__BoxShadowStyled-sc-rqe5sa-2 kKjtez hTNOqC"
    display="flex"
    width="110"
  >
    <button
      class="ButtonStyled-sc-1ys9v52-0 hXYavA"
      cursor="pointer"
      data-testid="edit"
    >
      Delete
    </button>
  </div>
</DocumentFragment>
`;
